### ISSUE TITLE
recursive removal of keys?

### ISSUE THREAD

**Issue by BernhardDenner** [ElektraInitiative/puppet-libelektra#1]
What about a 'remove_recursive' option for `kdbkey` to allow removal of the key and all subkeys if `ensure => absent`?

### Use case
allows removal of whole sections at once

**Comment by markus2330** [comment]
Might be useful to get rid of predefined sections. But take care that the API does not get inconsistent (between absent and present). How can someone specify that one key is present, but all others within the section are absent?

**Comment by BernhardDenner** [comment]
> How can someone specify that one key is present, but all others within the section are absent?

This is currently not possible and probably never might be.

It would require to something like 
```puppet
kdbkey { '.../path/to/section':
  ensure => present,
  purge_unmanaged_subkeys => true
}
# not removed since its a managed key
kdbkey { '.../path/to/section/setting':
  ensure => present,
  value   => 'xy'
}
```
The section key has to be aware of an existing 'managed' key and requires to check the catalog if such a child resource exists. I'm not sure if that can be done in an easy and good way.

However, I would consider this as an edge case ;)

**Comment by markus2330** [comment]
> However, I would consider this as an edge case ;)

Yes, but such edge cases might lead to confusing behaviour and cause more harm than good. If you do not find an elegant and expected way how to handle such conflicts, its better to not provide the feature.

How is it handled in the filesystem? Does puppet allow recursive removal of files?

What about a plugin that allows to write out configuration in `kdbkey` syntax?
Then someone who wants to get rid of a section, can simply export the keys and copy&paste them into the puppet manifest.

**Comment by BernhardDenner** [comment]
> Yes, but such edge cases might lead to confusing behaviour and cause more harm than good. If you do not find an elegant and expected way how to handle such conflicts, its better to not provide the feature.

Yes, its definitely a topic to think about before implementing such a feature.

> How is it handled in the filesystem? Does puppet allow recursive removal of files?

Yes, there's an option for that, while **keeping** managed files. So probably a hint on how to implement this ;)

> What about a plugin that allows to write out configuration in kdbkey syntax?

Might be an option too
